moit-domain/src/main/kotlin/com/mashup/moit/domain/moit/MoitService.kt

```diff
@@ -70,11 +70,7 @@ class MoitService(
     }
 
     fun getMoitByInvitationCode(invitationCode: String): Moit {
-        val moit = (moitRepository.findByInvitationCode(invitationCode.uppercase(Locale.getDefault()))
-            ?: throw MoitException.of(MoitExceptionType.NOT_EXIST))
-
-        if (moit.isEnd) throw MoitException.of(MoitExceptionType.INVALID_ACCESS, "종료된 Moit 입니다")
-
-        return moit.toDomain()
+        return moitRepository.findByInvitationCode(invitationCode.uppercase(Locale.getDefault()))
+            ?.validateDateForJoin()?.toDomain() ?: throw MoitException.of(MoitExceptionType.NOT_EXIST)
```

Comment:
`validateDateForJoin()` 가 검증하고 MoitEntity 를 반환하는 것이 살짝 어색하다는 느낌이 들기도 하는데 `validateDateForJoin()` 가 Unit 을 응답하도록 변경한다면 아래와 같이도 작성해 볼수도 있겠군요..!!

```suggestion
        return moitRepository.findByInvitationCode(invitationCode.uppercase(Locale.getDefault()))
            ?.also { it.validateDateForJoin() }?.toDomain()
            ?: throw MoitException.of(MoitExceptionType.NOT_EXIST)
```
```suggestion
        val moit = moitRepository.findByInvitationCode(invitationCode.uppercase(Locale.getDefault()))
            ?: throw MoitException.of(MoitExceptionType.NOT_EXIST)
        
        return moit.also { it.validateDateForJoin() }.toDomain()
```

바꿀 필요는 없고 그냥 의견 정도 입니다! 👍

Reply:
저도 validate 함수에서 자기 자신을 반환하는게 조금 어색해 보이네요.
also 써도 괜찮은데 저는 개인적으로 아예 체이닝에서 분리해도 좋을 것 같아요
```suggestion
        val moit = moitRepository.findByInvitationCode(invitationCode.uppercase(Locale.getDefault()))
        moit?.validateDateForJoin()
        return moit?.toDomain() ?: throw MoitException.of(MoitExceptionType.NOT_EXIST)
```